lux/action/similarity.py

```diff
@@ -39,11 +39,21 @@ def similar_pattern(ldf, intent, topK=-1):
     recommendations : Dict[str,obj]
         object with a collection of visualizations that result from the Similarity action
     """
-    row_specs = list(filter(lambda x: x.value != "", intent))
-    if len(row_specs) == 1:
-        search_space_vc = VisList(ldf.current_vis.collection.copy(), ldf)
 
-        query_vc = VisList(intent, ldf)
+    last = ldf.intent[-1]
```

Comment:
Why are we picking the last clause in the intent? There is no guarantee that the last element is the one with the value specified. You should fetch the clause by using `utils.get_filter_specs(intent)`.